src/android/SimpleCameraPreview.java

```diff
@@ -144,13 +158,17 @@ public void onRequestPermissionResult(int requestCode, String[] permissions, int
     }
 
     public void fetchLocation() {
-        if (ContextCompat.checkSelfPermission(cordova.getActivity(), android.Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
-            if (locationManager == null)
-                locationManager = (LocationManager) cordova.getActivity().getSystemService(Context.LOCATION_SERVICE);
-            Location cachedLocation = locationManager.getLastKnownLocation(LocationManager.NETWORK_PROVIDER);
-            if (cachedLocation != null)
-                fragment.setLocation(cachedLocation);
-            locationManager.requestLocationUpdates(LocationManager.NETWORK_PROVIDER, 0, 0, mLocationCallback);
+        try {
+            if (ContextCompat.checkSelfPermission(cordova.getActivity(), android.Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
+                if (locationManager == null)
+                    locationManager = (LocationManager) cordova.getActivity().getSystemService(Context.LOCATION_SERVICE);
+                Location cachedLocation = locationManager.getLastKnownLocation(LocationManager.NETWORK_PROVIDER);
+                if (cachedLocation != null)
+                    fragment.setLocation(cachedLocation);
+                locationManager.requestLocationUpdates(LocationManager.NETWORK_PROVIDER, 0, 0, mLocationCallback);
+            }
+        } catch(Exception e) {
+            throw e;
         }
```

Comment:
@dinitri-ragoo why changing this function ? Is there any reason to catch and then re-throw ?

Reply:
in case get `getSystemService `throws an exception